src/nv_ingest/modules/storages/embedding_storage.py

```diff
@@ -0,0 +1,159 @@
+from nv_ingest.util.modules.config_validator import fetch_and_validate_module_config
+from nv_ingest.util.tracing import traceable
+
+logger = logging.getLogger(__name__)
+
+MODULE_NAME = "embedding_storage"
+MODULE_NAMESPACE = "nv_ingest"
+
+# TODO: Move these into pipeline.py to populate the stage and validate them using the pydantic schema on startup.
+_DEFAULT_ENDPOINT = os.environ.get("MINIO_INTERNAL_ADDRESS", "minio:9000")
+_DEFAULT_READ_ADDRESS = os.environ.get("MINIO_PUBLIC_ADDRESS", "http://minio:9000")
+_DEFAULT_BUCKET_NAME = os.environ.get("MINIO_BUCKET", "nv-ingest")
+
+EmbeddingStorageLoaderFactory = ModuleLoaderFactory(MODULE_NAME, MODULE_NAMESPACE, EmbeddingStorageModuleSchema)
+
+def upload_embeddings(df: pd.DataFrame, params: Dict[str, Any]) -> pd.DataFrame:
+    """
+    Identify contents (e.g., images) within a dataframe and uploads the data to MinIO.
+    The image metadata in the metadata column is updated with the URL of the uploaded data.
+    """
+    
+    access_key = params.get("access_key", None)
+    secret_key = params.get("secret_key", None)
+
+    content_types = params.get("content_types")
+    endpoint = params.get("endpoint", _DEFAULT_ENDPOINT)
+    bucket_name = params.get("bucket_name", _DEFAULT_BUCKET_NAME)
+    bucket_path = params.get("bucket_path", "embeddings")
+
+    client = Minio(
+        endpoint,
+        access_key=access_key,
+        secret_key=secret_key,
+        session_token=params.get("session_token", None),
+        secure=params.get("secure", False),
+        region=params.get("region", None),
+    )
+
+    bucket_found = client.bucket_exists(bucket_name)
+    if not bucket_found:
+        client.make_bucket(bucket_name)
+        logger.debug("Created bucket %s", bucket_name)
+    else:
+        logger.debug("Bucket %s already exists", bucket_name)
+
+    storage_options = {
+        "key":access_key,
+        "secret": secret_key,
+        "client_kwargs": {
+        "endpoint_url": _DEFAULT_READ_ADDRESS
+        }
+    }
+
+    meta = df["metadata"]
+    emb_df_list = []  
+    file_uuid = uuid.uuid4().hex
+    destination_file = f"{bucket_path}/{file_uuid}.parquet"
+    write_path = f"s3://{bucket_name}/{destination_file}"
+    for idx, row in df.iterrows():
+        uu_id = row["uuid"]
+        metadata = row["metadata"].copy()
+
+        metadata["source_metadata"]["source_location"] = write_path
+
+        if row["document_type"] == ContentTypeEnum.EMBEDDING:
+            logger.debug("Storing embedding data to Minio")
+            metadata["embedding_metadata"][
+                "uploaded_embedding_url"
+            ] = write_path
+        # TODO: validate metadata before putting it back in.
+        # cm = str(metadata["content_metadata"])
+        text = metadata["content"]
+        # source_meta = str(metadata["source_metadata"])
+        emb = metadata["embedding"]
+        df.at[idx, "metadata"] = metadata
+
+        emb_df_list.append([emb, text])
```

Comment:
the portions of the record that are saved to this parquet are dependent on the schema of the collection in Milvus. We either need to make that schema static (as it is now) or we need to be able to consume that in this task when creating it.